src/RtspConnection.c

```diff
@@ -579,13 +579,20 @@ int performRtspHandshake(void) {
         }
 
         sessionId = getOptionContent(response.options, "Session");
-        if (sessionId == NULL) {
+
+        // Strips session id in the case the 
+        // RTSP server returns excess data.
+        // (i.e: DEADBEEF;timeout=)
+        char* session;
```

Comment:
Move this up with the other variable declarations in this block.

Reply:
Changed to use the original sessionId as strtok modifies it anyway (still won't affect functionality if there is no ";" - strtok leaves the string alone in this case).